statsmodels/discrete/discrete_model.py

```diff
@@ -502,22 +559,66 @@ def _derivative_exog(self, params, exog=None, transform='dydx',
 class MultinomialModel(BinaryModel):
 
     def _handle_data(self, endog, exog, missing, hasconst, **kwargs):
+        reference = kwargs.pop('reference', 0)
+
         if data_tools._is_using_ndarray_type(endog, None):
             endog_dummies, ynames = _numpy_to_dummies(endog)
+            keys = sorted(ynames.keys())
+            # sort the values
+            ynames = [_maybe_int(ynames[i]) for i in keys]
+            # if reference is an integer and
+            if not isinstance(reference, int):
+                reference_idx = _get_index_from_list(ynames, reference)
+            else:
+                reference_idx = reference
+                reference = ynames[reference]
             yname = 'y'
+            ynames = ["{0}={1}".format(yname, ynames[i]) for i in keys]
         elif data_tools._is_using_pandas(endog, None):
+            # patsy goes through here
             endog_dummies, ynames, yname = _pandas_to_dummies(endog)
```

Comment:
Not necessarily in this PR, but for pd.Categoricals a corner case to watch out for is when there are unused categories.